backend/workflow_manager/endpoint/source.py

```diff
@@ -154,36 +156,46 @@ def list_files_from_file_connector(self) -> tuple[dict[str, FileHash], int]:
         )
         root_dir_path = connector_settings.get(ConnectorKeys.PATH, "")
 
-        input_directory = str(source_configurations.get(SourceKey.ROOT_FOLDER, ""))
-
         source_fs = self.get_fs_connector(
             settings=connector_settings, connector_id=connector.connector_id
         )
-        input_directory = source_fs.get_connector_root_dir(
-            input_dir=input_directory, root_path=root_dir_path
-        )
-        logger.debug(f"source input directory {input_directory}")
-        if not isinstance(required_patterns, list):
-            required_patterns = [required_patterns]
-
         source_fs_fsspec = source_fs.get_fsspec_fs()
-
         patterns = self.valid_file_patterns(required_patterns=required_patterns)
-        is_directory = source_fs_fsspec.isdir(input_directory)
-        if not is_directory:
-            raise InvalidInputDirectory()
-        matched_files, count = self._get_matched_files(
-            source_fs_fsspec, input_directory, patterns, recursive, limit
+
+        folders_to_process = list(source_configurations.get(SourceKey.FOLDERS, ["/"]))
+        # Process from root in case its user provided list is empty
+        if not folders_to_process:
+            folders_to_process = ["/"]
+        logger.info(f"Folders to process: {folders_to_process}")
+        total_files_to_process = 0
+        total_matched_files = {}
+        for input_directory in folders_to_process:
+            input_directory = source_fs.get_connector_root_dir(
+                input_dir=input_directory, root_path=root_dir_path
+            )
+            logger.debug(f"Listing files from:  {input_directory}")
+            is_directory = source_fs_fsspec.isdir(input_directory)
+            if not is_directory:
```

Comment:
If the user provides multiple directories (e.g., a, b, c) and one of them (c) is not a valid directory, the current code will still process the valid directories (a and b) before raising an error for the invalid directory (c). This results in unnecessary processing for valid directories.

**Suggestions:** 
1. To avoid this, consider validating all provided directories upfront before processing any files. This ensures that we only proceed with file matching if all directories are valid, preventing wasted computation (especially for `_get_matched_files`).

2. Or Log the invalid directory and skip it, continuing with the remaining valid directories.

Reply:
I'm leaning towards 1 so that its visible to the user. Will perform this check outside the loop

Reply:
I am also lean towards suggestion 1.

Reply:
Implemented suggestion 1

Reply:
Need to either handle the error in the FE or raise it as an exception during WF run. Can be handled as a separate task